### ISSUE TITLE
Add an attachment to Contact Applicant Emails

### ISSUE THREAD

**Issue by lmullane** [bcgov/foi-flow#2698]
* As an IAO analyst
* I want to add attachment(s) to an applicant email
* so that I can attach letters created in AXIS

**Assumptions & Scope**	
What are the assumptions for this story?

IAO will need to send new fee estimates when a request is narrowed, the email failed or the applicant cannot find the email.

Letter templates will continue to be created in AXIS.

What is IN scope?

Attachments for applicant correspondence.

What is NOT in scope?

Sending the email is covered in #2696
Email templates for fees( new fee estimate or outstanding fees) are covered in #2687.
Applicant Correspondence Log is covered in #2684.
Other email or letter templates.

**Acceptance Criteria**	

**Scenario 1: Add an Attachment 1**
•	GIVEN the IAO analyst is on the Applicant Correspondence Log https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/941001462
•	AND the IAO analyst has selected an email template
•	WHEN the analyst clicks on the paper clip icon
* THEN the upload attachment modal will open

**Scenario 2: Drag and Drop a valid file type**
* GIVEN an IAO analyst is on the file upload modal: 
* WHEN I drag and drop a single pdf, Word or Excel file under 50 MB
* THEN the filename will be visible within the Drag and Drop space
* AND the save change button will be enabled

**Scenario 3: Drag and Drop an invalid file type**
* GIVEN a Ministry Coordinator is on the file upload modal
* WHEN I drag and drop a single file that is not a pdf, Word or Excel file, or is greater than 50 MB
* THEN the user will receive an error message stating the selected file type is invalid
* AND the Save Change button will remain disabled 

**Scenario 4: Click 'Add Files' button to browse files**
* GIVEN a Ministry Coordinator is on the file upload modal
* WHEN I click the 'Add Files' button 
* THEN a window will open to browse files on my local machine

**Scenario 5: Upload a valid file type using 'Add Files'**
* GIVEN I have selected 'Add Files' from the upload modal
* WHEN I select a single pdf, Word or Excel file under 50 MB
* AND select 'Ok'
* THEN the browse files window will close
* AND the filename will be visible within the Drag and Drop space
* AND the Save Change button will be enabled

**Scenario 6: Upload an invalid file type using 'Add Files'**
* GIVEN I have selected 'Add Files' from the upload modal
* WHEN I select a file that is not a pdf, Word or Excel file, or is greater than 50 MB
* THEN the user will receive an error message stating the selected file type is invalid
* AND the Save Change button will remain disabled 

**Scenario 7: Cancel upload of a valid file type using 'Add Files'**
* GIVEN I have selected to add a single valid file type under 50 MB from the browse files window
* WHEN I select 'Cancel'
* THEN the browse files window will close
* AND no filenames will be visible within the Drag and Drop space
* AND the Save Change button will remain disabled

**Scenario 8: Cancel Back to Review Request View"**
* GIVEN a Ministry Coordinator is on the upload file modal
* WHEN I select 'Cancel'
* THEN the upload file modal will close, and the request state will remain as it was prior to opening the modal

**Scenario 9: Activate the Send Email button**
* GIVEN a Ministry Coordinator is on the upload file modal
* WHEN a valid filetype has been uploaded
•	THEN the attachment will be uploaded to the email
•	AND the title of the attachment will be displayed
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/941001462
*     AND the blue 'Send Email' button will become blue and active

**Scenario 10: Remove an Attachment **
•	GIVEN the IAO analyst has added an attachment to the email
•.    WHEN  the IAO analyst clicks the 'x' next to the PDF document 
•	THEN the attachment will be removed

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by lmullane** [comment]
Follows same pattern as other file uploads.

**Comment by KaraBeach** [comment]
@nkan-aot - all A/C's have passed. @lmullane @JHarrietha-AOT  Just confirming design wise that the button to save changes currently reads "continue" rather than "save or save changes" as it is listed in the AC.  Is this correct?

**Comment by JHarrietha-AOT** [comment]
@nkan-aot Can we update the button to say Save Changes? Not sure how long this little change would take. But @KaraBeach we could just wait for @nkan-aot to push that update, then double check it when Nick tags to say it has been changed and it can be pushed to close then.

**Comment by nkan-aot** [comment]
@JHarrietha-AOT sure I will let you both know once that is done

**Comment by nkan-aot** [comment]
@JHarrietha-AOT sure I will let you both know once that is done

**Comment by KaraBeach** [comment]
@nkan-aot - thanks for fixing the button to "save changes" closing as all ACs pass.
